machine.go

```diff
@@ -98,6 +98,33 @@ type Config struct {
 	JailerCfg JailerConfig
 }
 
+// WithJailerConfig will set the given JailerConfig to the JailerCfg field. In
+// addition this will also set EnableJailer to true
```

Comment:
Do we strictly need to have the `EnableJailer` field or can we enable the jailer when `JailerCfg != nil`?

Reply:
Yea, that is an option, using nil as an indicator. I wanted to prevent doing that since whenever you want to grab values from the jailer config, you'd need to nil check. But I am okay with using nil as a check if we think that is a better experience.

Reply:
I agree it seems simpler to just have our code do nil checks internally and enable the jailer when JailerCfg is not nil. I think I'd be confused as a user if I saw there was `EnableJailer`, `JailerCfg` and `WithJailerConfig`.